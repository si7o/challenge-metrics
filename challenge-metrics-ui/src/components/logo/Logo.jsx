import React from 'react'
import './styles.css'
import logo from './logo.svg'

const Logo = (props) => {
  return (
    <img src={logo} className='logo' alt='logo' />
  )
}

export default Logo
