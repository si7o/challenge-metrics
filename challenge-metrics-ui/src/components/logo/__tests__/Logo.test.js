import React from 'react'
import { render, screen } from '@testing-library/react'

import Logo from '../Logo'

describe('<Logo>', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('renders default', () => {
    const { container } = render(<Logo />)

    expect(container).toMatchSnapshot()
  })

  test('size large sets "large" classname', () => {
    render(<Logo size='large' />)

    expect(screen.getByRole('img')).toHaveClass('large')
  })

  test('spin fast sets "fast" classname', () => {
    render(<Logo spin='fast' />)

    expect(screen.getByRole('img')).toHaveClass('fast')
  })
})
