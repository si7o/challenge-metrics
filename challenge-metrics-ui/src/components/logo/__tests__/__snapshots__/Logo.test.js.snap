// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> renders default 1`] = `
<div>
  <img
    alt="logo"
    class="logo default normal"
    src="logo.svg"
  />
</div>
`;
